src/diff/children.js

```diff
@@ -119,7 +120,10 @@ export function diffChildren(
 		}
 		// If this node suspended during hydration, and no other flags are set:
 		// @TODO: might be better to explicitly check for MODE_ERRORED here.
-		else if ((childInternal._flags ^ (MODE_HYDRATE | MODE_SUSPENDED)) === 0) {
```

Comment:
@developit This change didn't work for me cause the result is never `0`. When both `MODE_HYDRATE` and `MODE_SUSPENDED` flags are set, the result is all other flags set in `_flags`.

Here is a simplified example. Imagine the left most bit is a flag that represents the type of the internal:

| \_flags | Hydrate \| Suspended | Result with ^ | What we want |
| ------- | -------------------- | ------------- | ------------ |
| 100     | 011                  | 111           | false        |
| 101     | 011                  | 110           | false        |
| 110     | 011                  | 101           | false        |
| 111     | 011                  | 100           | true         |

Reply:
What you ended up with here makes more sense to me anyway and I believe will also be fastest (x & 6 === 6).